### PR TITLE
#9Task.PopovDmitriy

### PR DESCRIPTION
**Pull request by PopovDmitriy**
https://travis-ci.com/github/PopovDmitriy/js-assignments/branches

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -104,7 +115,16 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+    return () => {
+        while (attempts > 0) {
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js]: use for-loop for loops with known number of iterations

**Review by AlexMokin — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -104,7 +115,16 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+    return () => {
+        for (let i = 0; i < attempts;) {
```
Comment (AlexMokin) [inline task/09-functions-n-closures-tasks.js]: looks like while-loop with extra steps. You added **i** variable but it's always 0, what's the point of using it?

Try to rewrite it like **for (let i = 0; i < attempts; i++)** and without modifiying **attempts**

**Review by AlexMokin — COMMENTED** [review]
Check a new comment. Also PR must contain only changes for a given task

I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/09-functions-n-closures-tasks.js (hunk 2 above)
Comment (PopovDmitriy) [reply task/09-functions-n-closures-tasks.js]: I think something like this

**Comment by IlyaMokin** [comment]
Closing as accepted!

**Pull request closed without merging**